### PR TITLE
Execute order

### LINKED ISSUE #44: Change the demo app start script to use expo start -c
Thoughts on whether this is a good idea?

### PR DESCRIPTION
**Pull request by marcelomorgado**
### Issue link

Refs: #62, #59, #20
Closes #44 

### Types of changes

New feature (non-breaking change that adds functionality)

Meta (a change that affects future changes to this repo)

Technical debt (a code change that doesn't fix a bug or add a feature but makes something clearer for devs)

### Notes

- Functional implementation focused,
- Should blockchain operations functions be moved to `redux` action functions?
- Loading and errors alerts are missing for now (todo).

### CODE REVIEW

**Comment by pcowgill** [comment]
@marcelomorgado Great work! I'll review this ASAP. Thanks!

**Comment by pcowgill** [comment]
> * Functional implementation focused,

Do you mean you're focused on getting something working as quickly as possible? Or using a functional programming style as much as possible?

> * Should blockchain operations functions be moved to `redux` action functions?

Hmm, yes, I think so. I'm a little tempted to say the demo app should be without the redux dep. I think in the long run we'll likely want a demo app without redux too though, especially with the direction React is going these days.

In the Decentraland app I think we'll definitely want redux. At what point do  you think we should stop adding things to the demo app and move over to the `decentraland` dir and extract things to the `shared` folder? I would say it's still too premature for that.

So in short, my vote is to do this with redux thunk or saga, yes.

> * Loading and errors alerts are missing for now (todo).

Noted, thanks. Please put TODOs in the code wherever appropriate too. Should we track this in an issue?

**Comment by marcelomorgado** [comment]
> > * Functional implementation focused,
> 
> Do you mean you're focused on getting something working as quickly as possible? Or using a functional programming style as much as possible?

Oh, I've meant focused on getting something working as quickly as possible / less UX focused.


> > * Should blockchain operations functions be moved to `redux` action functions?
> 
> Hmm, yes, I think so. I'm a little tempted to say the demo app should be without the redux dep. I think in the long run we'll likely want a demo app without redux too though, especially with the direction React is going these days.
> 
> In the Decentraland app I think we'll definitely want redux. At what point do you think we should stop adding things to the demo app and move over to the `decentraland` dir and extract things to the `shared` folder? I would say it's still too premature for that.
> 
> So in short, my vote is to do this with redux thunk or saga, yes.

Hmm, Ok, I've added that because of issue #20 and because I've started needed to pass a lot of props down thru a lot of components. Is the `React context API` the best way to go in midterm?
Maybe when the app be less unstable? Or when that be good enough to a presentation? 

> > * Loading and errors alerts are missing for now (todo).
> 
> Noted, thanks. Please put TODOs in the code wherever appropriate too. Should we track this in an issue?
Got it, I'll comment TODOs on code whenever is possible.
I was waiting your response here, I've just created: https://github.com/tasitlabs/tasit/issues/95

**Comment by marcelomorgado** [comment]
> > > * Functional implementation focused,
> > 
> > 
> > Do you mean you're focused on getting something working as quickly as possible? Or using a functional programming style as much as possible?

I think that I shoud have said "back-end focused"

File: demo/screens/SellOrderClaimScreen.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { claimSellOrder } from "../actions";
+import SellOrderClaim from "@presentational/SellOrderClaim";
+
+export class SellOrderClaimScreen extends React.Component {
+  _onClaim = sellOrder => {
+    const { claimSellOrder } = this.props;
+
+    claimSellOrder(sellOrder);
```
Comment (pcowgill) [inline demo/screens/SellOrderClaimScreen.js]: Is this function sync or async?

File: demo/screens/SellOrderExecuteScreen.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import { connect } from "react-redux";
+import { removeSellOrder } from "../actions";
+import SellOrderExecute from "@presentational/SellOrderExecute";
+import PropTypes from "prop-types";
+import ContractsABIs from "@constants/ContractsABIs";
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]:
```suggestion

import ContractsABIs from "@constants/ContractsABIs";
```

File: demo/screens/SellOrderExecuteScreen.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+const { manaABI, estateABI, marketplaceABI } = ContractsABIs;
+import ContractsAddresses from "@constants/ContractsAddresses";
+const { manaAddress, estateAddress, marketplaceAddress } = ContractsAddresses;
+import { Action } from "tasit-sdk";
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]:
```suggestion

import { Action } from "tasit-sdk";
```

File: demo/screens/SellOrderExecuteScreen.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+const { Contract } = Action;
+import { createFromPrivateKey } from "tasit-account/dist/testHelpers/helpers";
+
+export class SellOrderExecuteScreen extends React.Component {
+  // TODO: Switch to new DecentralandEstate() once the SDK includes that
+  estateContract = new Contract(estateAddress, estateABI);
+  marketplaceContract = new Contract(marketplaceAddress, marketplaceABI);
+  manaContract = new Contract(manaAddress, manaABI);
+
+  _onOrderExecution = sellOrder => {
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]: I'm wondering whether we can find a less fintech-like name for `sellOrder`

File: demo/screens/SellOrderExecuteScreen.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    const { navigation } = this.props;
+    navigation.navigate("ListSellOrdersScreen");
+    this._executeOrder(sellOrder);
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]: Do we want to do `_executeOrder` after the navigation? I think we probably do, but I want to make sure I understand the rationale. Thanks!

File: demo/screens/SellOrderExecuteScreen.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  };
+
+  _executeOrder = async sellOrder => {
+    const { account, removeSellOrder } = this.props;
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]: I think I'd like to get the props outside of this function so that the function itself is purely functional and has these explicitly as parameters. Thoughts?

File: demo/screens/SellOrderExecuteScreen.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    // See: https://github.com/tasitlabs/TasitSDK/issues/173
+    const gasParams = {
+      gasLimit: 7e6,
+      gasPrice: 1e9,
+    };
+
+    const marketplaceAddress = this.marketplaceContract.getAddress();
+    const estateAddress = this.estateContract.getAddress();
+
+    await this._manaFaucetTo(account, TEN);
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]: Should we do this outside of this screen?

File: demo/screens/SellOrderExecuteScreen.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+    await this._approveManaSpending(account, marketplaceAddress, ONE);
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]:
```suggestion
    // TODO: Move this out to an explicit approval flow?
    await this._approveManaSpending(account, marketplaceAddress, ONE);
```

File: demo/screens/SellOrderExecuteScreen.js [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+      estateAddress,
+      asset.id,
+      priceInWei.toString(),
+      fingerprint.toString(),
+      gasParams
+    );
+
+    await executeOrder.waitForNonceToUpdate();
+
+    // TODO: This function should be called inside of the eventLister
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]:
```suggestion
    // TODO: This function should be called inside of the eventListener
```

File: demo/screens/SellOrderExecuteScreen.js [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+    // that catch the safeExecuteOrder successfull event.
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]:
```suggestion
    // that catches the safeExecuteOrder successful event.
```

File: demo/screens/SellOrderExecuteScreen.js [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+    removeSellOrder(sellOrder);
+  };
+
+  _approveManaSpending = async (fromAccount, toAddress, value) => {
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]: Maybe extract this function out of this screen and import it on this screen? That would make it easier to move around to different components / screens in the future too.

File: demo/screens/SellOrderExecuteScreen.js [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+    return (
+      <SellOrderExecute
+        sellOrder={sellOrder}
+        onOrderExecution={() => this._onOrderExecution(sellOrder)}
+      />
+    );
+  }
+}
+
+SellOrderExecuteScreen.propTypes = {
```
Comment (pcowgill) [inline demo/screens/SellOrderExecuteScreen.js]: From the buyer's perspective it's a buy order rather than a sell order, right?

File: demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap [hunk 13]
```diff
@@ -0,0 +1,7 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`SellOrderClaimScreen renders the component 1`] = `
+<ContextConsumer>
```
Comment (pcowgill) [inline demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap]: This screen's snapshot is so small that the test isn't really testing much. Should we drill down on this screen more?

Comment (pcowgill) [reply demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap]: Then again, I guess the presentational components are really where the interesting snapshot tests are, right?

**Comment by pcowgill** [comment]
> I think that I shoud have said "back-end focused"

Ah okay, makes sense!

**Comment by pcowgill** [comment]
> Hmm, Ok, I've added that because of issue #20 and because I've started needed to pass a lot of props down thru a lot of components. Is the `React context API` the best way to go in midterm?
Maybe when the app be less unstable? Or when that be good enough to a presentation?

I think having it as a dep for now is a great next step. Thanks for including it. We have plenty of good options for the medium term, and regardless this is a step in the right direction. Introducing redux when a lot of props are being passed down is a great reason.

Yeah, I think React context API is worth considering in the medium term. Let's included it in issue #20. Let's stick with what we have while the app is under constant change.

Can you rephrase "Or when that be good enough to a presentation?"? Thanks!

> Got it, I'll comment TODOs on code whenever is possible.
I was waiting your response here, I've just created: #95

Perfect, thanks

**Comment by pcowgill** [comment]
I think this could say Closes #44 

Oh, maybe since listening is decoupled from the writing of data here, async support with thunk or saga is less relevant?

File: demo/AppNavigator.js [hunk 14]
```diff
@@ -2,7 +2,8 @@ import { createStackNavigator, createAppContainer } from "react-navigation";
 
 import HomeScreen from "./screens/HomeScreen";
 import ListSellOrdersScreen from "./screens/ListSellOrdersScreen";
-import SellOrderScreen from "./screens/SellOrderScreen";
+import SellOrderClaimScreen from "./screens/SellOrderClaimScreen";
```
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/AppNavigator.js]: OK. I'll change to same as "Buy" screen with a different flow when users have no tokens yet. Make sense?

File: demo/components/presentational/EthereumSignUp.test.js [hunk 15]
```diff
@@ -4,6 +4,6 @@ import EthereumSignUp from "./EthereumSignUp";
 
 describe("EthereumSignUp", () => {
   it("renders the component", async () => {
-    expect(shallow(<EthereumSignUp />)).toMatchSnapshot();
+    expect(shallow(<EthereumSignUp onSignUp={() => {}} />)).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/components/presentational/EthereumSignUp.test.js]: For me is the same in that case. But having named functions is better when the function does something. So, for the project code style standard make sense always have the functions as variables. I'll do that.

File: demo/components/presentational/EthereumSignUpForm.js [hunk 16]
```diff
@@ -1,26 +1,18 @@
...
   responsiveWidth,
   responsiveFontSize,
 } from "react-native-responsive-dimensions";
-import { Account } from "tasit-sdk";
+import PropTypes from "prop-types";
 import Button from "./Button";
 
 export default class EthereumSignUpForm extends React.Component {
-  state = {
-    text: "",
```
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/components/presentational/EthereumSignUpForm.js]: I can store it, but for now, that isn't be used.
And it is a question that I was going to do, what the best way to store fields from form components? I saw both redux / local state as well.

File: demo/reducers.js [hunk 17]
```diff
@@ -0,0 +1,47 @@
+import { combineReducers } from "redux";
+import {
+  SET_ACCOUNT,
+  CLAIM_SELL_ORDER,
+  SET_SELL_ORDERS,
+  REMOVE_SELL_ORDER,
+} from "./actions";
+
+function account(state = null, action) {
```
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/reducers.js]: Please, correct if I'm wrong but I'm trying to keep reducers name same as `state` vars names.
Or is better do something like this?

```javascript
const demoApp = combineReducers({
  account: setUserAccount,
  claimedSellOrder,
  sellOrders,
});
```

File: demo/reducers.js [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+      return sellOrder;
+    default:
+      return state;
+  }
+}
+
+function sellOrders(state = [], action) {
+  const { type, sellOrders, sellOrder } = action;
+  switch (type) {
+    case SET_SELL_ORDERS:
```
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/reducers.js]: Yes, I think so or/and `APPEND_SELL_ORDERS` depending on how the fetch logic will be.

File: demo/screens/EthereumSignUpScreen.js [hunk 19]
```diff
@@ -1,8 +1,43 @@
...
+EthereumSignUpScreen.propTypes = {
+  setAccount: PropTypes.func.isRequired,
+};
+
+const mapDispatchToProps = {
+  setAccount,
+};
+
+export default connect(
+  null,
```
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/screens/EthereumSignUpScreen.js]:
```javascript
connect(mapStateToProps, mapDispatchToProps)
``` 
There is a better way to do that? Should I create `const mapStateToProps = null` and use it?

File: demo/AppNavigator.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/AppNavigator.js]: Hmm, I'm not sure I follow.

I guess it depends whether we're talking about buying ERC20 tokens or buying the ERC721 estate.

I think whether or not they have tokens, they're still hoping to buy the ERC721 estate, right?

File: demo/screens/SellOrderClaimScreen.js (hunk 1 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderClaimScreen.js]: sync this dispatch functions is used to store which land users want to buy.

File: demo/components/presentational/EthereumSignUp.test.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/components/presentational/EthereumSignUp.test.js]: Okay, that makes sense. Sounds good. Thanks!

File: demo/components/presentational/EthereumSignUpForm.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/components/presentational/EthereumSignUpForm.js]: I've heard arguments that form fields should always store state locally unless there's any chance of another component using that state in the future.

True though, we aren't using it yet, so it doesn't matter much either way right now

File: demo/reducers.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/reducers.js]: Hmm, that makes sense. Never mind - I like the way you have it now. Thanks! Resolving this.

File: demo/reducers.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/reducers.js]: Yep. Okay, just checking. Resolving this for now.

File: demo/screens/EthereumSignUpScreen.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/screens/EthereumSignUpScreen.js]: Hmm, I think it's fine the way you have it now or with the alternative you suggested - up to you. I was asking mostly just to remind myself.

File: demo/screens/SellOrderClaimScreen.js (hunk 1 above)
Comment (pcowgill) [reply demo/screens/SellOrderClaimScreen.js]: Okay, thanks. Resolving this.

File: demo/AppNavigator.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/AppNavigator.js]: Sorry.
Yes. From UI point of view, I said to change the 'Claim' button to 'Buy' button on Estate Sell Order Screen and then click to buy, the user will follow the SignUp flow.

File: demo/screens/SellOrderExecuteScreen.js (hunk 4 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: The `Marketplace` contract calls as just `order`, do you prefer?

File: demo/screens/SellOrderExecuteScreen.js (hunk 5 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Since blockchain operations are slow, I've thought that make the user go ahead and show
a loading/progress feedback (not implemented yet) would be better than keeping waiting. But I'm not sure. What do you think?

File: demo/screens/SellOrderExecuteScreen.js (hunk 6 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Yes, make sense.

File: demo/screens/SellOrderExecuteScreen.js (hunk 7 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Could be. I've tried to implement the screens/flows how they are now.
I think that will change after we decided how 0.1.0 screens/flows would be (https://github.com/tasitlabs/tasit/issues/54). Make sense?

File: demo/screens/SellOrderExecuteScreen.js (hunk 11 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Ok, any suggestion of how is the best way to do that? A `helper.js` file?

File: demo/screens/SellOrderExecuteScreen.js (hunk 5 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: I agree - that might be worth putting in the code as a note.

Could we also achieve that by dispatching an async action before navigating to the new screen?

Thanks!

File: demo/screens/SellOrderExecuteScreen.js (hunk 4 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Hmm, I think that's still targets a financial audience. Maybe `LandForSale` / `EstateForSale`?

File: demo/screens/SellOrderExecuteScreen.js (hunk 12 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: I'm not sure. Using Decentraland Marketplace terms, it is an order to be executed.
There is no 'buy order', buyers only execute existing orders created by sellers.

File: demo/AppNavigator.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/AppNavigator.js]: Ah right. I was confused because we have a different thread about rewording the "sell order" part.

Sure, that sounds good!

File: demo/screens/SellOrderExecuteScreen.js (hunk 12 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: True. So maybe if we removed the word order it would free up our language. I don't necessarily want to use the language from the contracts on the front end if there's simpler language we can find.

Maybe `BuyLandScreen` or `LandCheckoutScreen`?

File: demo/screens/SellOrderExecuteScreen.js (hunk 11 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Sure. helpers.js sounds good. Then if we find any common themes between subsets of the helpers, we can split those out into files with more specific names accordingly

File: demo/screens/SellOrderExecuteScreen.js (hunk 7 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Yeah, good point. We'll keep it here now until the whole flow is solidified.

Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Resolving this

File: demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap (hunk 13 above)
Comment (marcelomorgado) [reply demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap]: This test is small because that is importing `redux` connected HoC. Snapshot is better now after fix. I'll review all tests/snapshots.

**Comment by marcelomorgado** [comment]
> > Hmm, Ok, I've added that because of issue #20 and because I've started needed to pass a lot of props down thru a lot of components. Is the `React context API` the best way to go in midterm?
> > Maybe when the app be less unstable? Or when that be good enough to a presentation?
> 
> I think having it as a dep for now is a great next step. Thanks for including it. We have plenty of good options for the medium term, and regardless this is a step in the right direction. Introducing redux when a lot of props are being passed down is a great reason.

:+1: 

> Yeah, I think React context API is worth considering in the medium term. Let's included it in issue #20. Let's stick with what we have while the app is under constant change.

Done


> Can you rephrase "Or when that be good enough to a presentation?"? Thanks!

Oh, sorry: When the app is good/stable enough to be used on presentations about `tasit?`. 


> > Got it, I'll comment TODOs on code whenever is possible.
> > I was waiting your response here, I've just created: #95
> 
> Perfect, thanks

:+1:

**Comment by marcelomorgado** [comment]
> I think this could say Closes #44

Yes, since #44 is about `expo start -c`.

> Oh, maybe since listening is decoupled from the writing of data here, async support with thunk or saga is less relevant?

I'm not sure about `thunk`/`saga`. But because I've added a comment to use React context API I think that make sense keep it open (#20).

**Comment by pcowgill** [comment]
>> Can you rephrase "Or when that be good enough to a presentation?"? Thanks!

> Oh, sorry: When the app is good/stable enough to be used on presentations about `tasit?`.

Ah okay yes, that's the point where we'll want to start considering potential tech debt issues like this. (And it may not even turn out to be tech debt how we have it now!)

File: demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap (hunk 13 above)
Comment (pcowgill) [reply demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap]: Makes sense. Sounds good!

File: demo/components/presentational/EthereumSignUp.test.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/components/presentational/EthereumSignUp.test.js]: Done.

File: demo/components/presentational/EthereumSignUpForm.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/components/presentational/EthereumSignUpForm.js]: Ok, I'll keep that way now.

File: demo/screens/EthereumSignUpScreen.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/screens/EthereumSignUpScreen.js]: Ok, I'll keep that way now.

File: demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap (hunk 13 above)
Comment (marcelomorgado) [reply demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap]: Done.

File: demo/screens/SellOrderExecuteScreen.js (hunk 6 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Done.

File: demo/components/presentational/EthereumSignUpForm.js (hunk 16 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/components/presentational/EthereumSignUpForm.js]: Resolving this

File: demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap (hunk 13 above)
Comment (pcowgill) [reply demo/screens/__snapshots__/SellOrderClaimScreen.test.js.snap]: Thanks! Resolving this

File: demo/screens/ListSellOrdersScreen.test.js [hunk 20]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import NavigationTestUtils from "react-navigation/NavigationTestUtils";
 import { shallow } from "enzyme";
-import ListSellOrdersScreen from "./ListSellOrdersScreen";
+import { ListSellOrdersScreen } from "./ListSellOrdersScreen";
```
Comment (pcowgill) [inline demo/screens/ListSellOrdersScreen.test.js]: Oh interesting, what led to this change being needed?

File: demo/components/presentational/EthereumSignUp.test.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/components/presentational/EthereumSignUp.test.js]: Thanks! Resolving

File: demo/screens/SellOrderExecuteScreen.js (hunk 11 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Done.

File: demo/screens/ListSellOrdersScreen.test.js (hunk 20 above)
Comment (marcelomorgado) [reply demo/screens/ListSellOrdersScreen.test.js]: Yes, it's an interesting trick. Since `redux` connected component exposes HoC, a way to bypass that and access the component directly is to export the class too. Example:

**Component**
```javascript
export class SomeComponent extends React.Component {...}
export default connect(mapStateToProps, mapDispatchToProps)(SomeComponent);
```
**Test**
```javascript
// Connected HoC
//import SomeComponent from "./SomeComponent";

// Component
import { SomeComponent } from "./SomeComponent";
```
Refs: https://redux.js.org/recipes/writing-tests#connected-components

File: demo/screens/SellOrderExecuteScreen.js (hunk 5 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Yeah, I've done that, since `_executeOrder` is already async.
Missing move that to a `redux` action.

File: demo/AppNavigator.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (marcelomorgado) [reply demo/AppNavigator.js]: Done.

File: demo/screens/SellOrderExecuteScreen.js (hunk 4 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Since the user can buy either `Estates` and `Parcels` I'm not sure but seem that the `Land` is a better generic name.

File: demo/screens/SellOrderExecuteScreen.js (hunk 5 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Seems that [redux docs](https://redux.js.org/advanced/async-actions) recommends keep `actions` as plain functions and use middleware (thunk/saga) to handle with fetching data. What do you think?

File: demo/AppNavigator.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/AppNavigator.js]: Thanks! Resolving

File: demo/screens/EthereumSignUpScreen.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (pcowgill) [reply demo/screens/EthereumSignUpScreen.js]: Okay, resolving!

File: demo/screens/SellOrderExecuteScreen.js (hunk 4 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Works for me! It’s still a little confusing because “land” (the contract for land parcels) is land and “estates” are land in English grammar. But I can’t think of a good way around that.

File: demo/screens/ListSellOrdersScreen.test.js (hunk 20 above)
Comment (pcowgill) [reply demo/screens/ListSellOrdersScreen.test.js]: Interesting! Thanks for explaining. Resolving.

File: demo/screens/SellOrderExecuteScreen.js (hunk 5 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Yeah, I think that’s the way to go

File: demo/screens/SellOrderExecuteScreen.js (hunk 11 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Thanks! Resolving

File: demo/screens/SellOrderExecuteScreen.js (hunk 4 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Done.

File: demo/screens/SellOrderExecuteScreen.js (hunk 5 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Can we extract this thread to #20 ?

File: demo/screens/SellOrderExecuteScreen.js (hunk 12 above)
Comment (marcelomorgado) [reply demo/screens/SellOrderExecuteScreen.js]: Done.

File: demo/screens/SellOrderExecuteScreen.js (hunk 4 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Thanks! Resolving this

File: demo/screens/SellOrderExecuteScreen.js (hunk 5 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Yep! Feel free to resolve this once you do.

File: demo/screens/SellOrderExecuteScreen.js (hunk 12 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Thanks! Resolving

File: demo/AppNavigator.js [hunk 21]
```diff
@@ -1,9 +1,8 @@
 import { createStackNavigator, createAppContainer } from "react-navigation";
 
 import HomeScreen from "./screens/HomeScreen";
-import ListSellOrdersScreen from "./screens/ListSellOrdersScreen";
-import SellOrderClaimScreen from "./screens/SellOrderClaimScreen";
-import SellOrderExecuteScreen from "./screens/SellOrderExecuteScreen";
+import ListLandsForSaleScreen from "./screens/ListLandsForSaleScreen";
```
Comment (pcowgill) [inline demo/AppNavigator.js]: The only problem is that land can't be plural

File: demo/screens/SellOrderExecuteScreen.js (hunk 6 above)
Comment (pcowgill) [reply demo/screens/SellOrderExecuteScreen.js]: Thanks! Resolving

**Comment by pcowgill** [comment]
@marcelomorgado I think we can merge this, right?

**Comment by marcelomorgado** [comment]
> @marcelomorgado I think we can merge this, right?

Right

File: demo/AppNavigator.js (hunk 21 above)
Comment (marcelomorgado) [reply demo/AppNavigator.js]: Hm... is `ListLandForSaleScreen` correct?

Comment (pcowgill) [reply demo/AppNavigator.js]: Yeah, that works. The “list” word still shows that it’s for multiple parcels
